Dockerfile

```diff
@@ -0,0 +1,12 @@
+FROM nginx:alpine
+EXPOSE 80
+ENV LISTEN_PORT=80 \
+  SERVER_HOST=otrecorder \
+  SERVER_PORT=80
+
+COPY nginx.tmpl /etc/nginx/nginx.tmpl
+COPY index.html /usr/share/nginx/html
+COPY static/ /usr/share/nginx/html/static/
+RUN ls /usr/share/nginx/html/*
+CMD /bin/sh -c "envsubst '\${SERVER_HOST} \${SERVER_PORT} \${LISTEN_PORT}' < /etc/nginx/nginx.tmpl > /etc/nginx/nginx.conf && nginx -g 'daemon off;' || cat /etc/nginx/nginx.conf"
```

Comment:
Same here for `cat /etc/nginx/nginx.conf`

Reply:
The cat is a trick I saw somewhere for helping in case nginx fails to start. This allows to understand quicker if something is wrong with nginx.conf like if an env variable is missing. IMHO it's OK to keep it in production, but I can remove it if you'd rather.

Reply:
I think that's fine, thanks for the explanation :)

Reply:
while it prints `nginx.conf` in case of failure it's not particularly useful IMO because the important bits such as the variables aren't shown. I think something like

```
|| ( env; cat /etc/nginx/nginx.conf )
```

could be more useful.

Reply:
If you talk about SERVER_HOST and SERVER_PORT , they are already shown because they are replaced by the `envsubst` just before. I believe it is enough ? Or am I missing something else ?

Reply:
The values will likely not be visible if the execution fails, and that’s the whole point of the `||` construction, isn’t it? :) My intention is to show the full environment upon failure.

Reply:
OK I see it now, there may be cases where it is more clear, thanks. I applied your suggestion.